build_scripts/build_macos_m1.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+set -euo pipefail
+
+pip install setuptools_scm
+# The environment variable CHIA_INSTALLER_VERSION needs to be defined.
+# If the env variable NOTARIZE and the username and password variables are
+# set, this will attempt to Notarize the signed DMG.
+CHIA_INSTALLER_VERSION=$(python installer-version.py)
+
+if [ ! "$CHIA_INSTALLER_VERSION" ]; then
+	echo "WARNING: No environment variable CHIA_INSTALLER_VERSION set. Using 0.0.0."
+	CHIA_INSTALLER_VERSION="0.0.0"
+fi
+echo "Chia Installer Version is: $CHIA_INSTALLER_VERSION"
+
+echo "Installing npm and electron packagers"
+npm install electron-installer-dmg -g
+npm install electron-packager -g
+npm install electron/electron-osx-sign -g
+npm install notarize-cli -g
+
+echo "Create dist/"
+sudo rm -rf dist
+mkdir dist
+
+echo "Install pyinstaller and build bootloaders for M1"
+#pip install pyinstaller==4.3
+# Once there is a 4.4, we can clone that tag and build that
+# M1 support isn't in a tag yet.
+# Alternatively, if the m1 bootloaders are distributed with pip in the future, can just use those
+git clone https://github.com/pyinstaller/pyinstaller.git
+cd pyinstaller/bootloader
+python ./waf all
+pip install ..
+cd ../..
+
+echo "Create executables with pyinstaller"
+SPEC_FILE=$(python -c 'import chia; print(chia.PYINSTALLER_SPEC_PATH)')
+pyinstaller --log-level=INFO "$SPEC_FILE"
+LAST_EXIT_CODE=$?
+if [ "$LAST_EXIT_CODE" -ne 0 ]; then
+	echo >&2 "pyinstaller failed!"
+	exit $LAST_EXIT_CODE
+fi
+cp -r dist/daemon ../chia-blockchain-gui
+cd .. || exit
+cd chia-blockchain-gui || exit
+
+echo "npm build"
+npm install
+npm audit fix
+npm run build
+LAST_EXIT_CODE=$?
+if [ "$LAST_EXIT_CODE" -ne 0 ]; then
+	echo >&2 "npm run build failed!"
+	exit $LAST_EXIT_CODE
+fi
+
+electron-packager . Chia --asar.unpack="**/daemon/**" --platform=darwin \
```

Comment:
Is electron packager ARM aware? A little worried about platform=Darwin here

Reply:
I think it must be - this whole script is run under the arch -arm64 flag and the resulting installer builds and installs on an m1.. Tested on m1-02 and was able to launch Chia and start syncing. I believe Roy tested on an M1 as well.. The main thing I had issues with was pyinstaller, but thats now resolved as well.